networkit/cpp/graph/test/GraphToolsGTest.cpp

```diff
@@ -101,6 +101,48 @@ TEST_P(GraphToolsGTest, testDensity) {
     }
 }
 
+TEST_P(GraphToolsGTest, testVolume) {
+    constexpr count n = 100;
+    constexpr double p = 0.1;
+
+    auto doTest = [&](const Graph &G) {
+        // Volume for graph G
+        const auto volume = GraphTools::volume(G);
+
+        // Volume for either directed/undirected
+        double mod = directed() ? 1.0 : 2.0;
+        double baseG = weighted() ? static_cast<double>(G.totalEdgeWeight())
+                                  : static_cast<double>(G.numberOfEdges());
+        EXPECT_NEAR(volume, mod * baseG, 1e-7);
+ 
+        // Volume for subgraph G2
+        node seed = Aux::Random::integer(G.upperNodeIdBound());
+        std::vector<node> sn(G.neighborRange(seed).begin(), G.neighborRange(seed).end());
+        auto sNodes = std::unordered_set<node>(sn.begin(), sn.end());
+        sNodes.insert(seed);
+        const auto G2 = GraphTools::subgraphFromNodes(G, sNodes);
+
+        const auto outVolumeSub = GraphTools::subgraphVolume(G, sNodes);
+        const auto inVolumeSub = GraphTools::subgraphInVolume(G, sNodes);
+
+        // Volume for either directed/undirected
+        double baseG2 = weighted() ? static_cast<double>(G2.totalEdgeWeight())
+                                   : static_cast<double>(G2.numberOfEdges());
+        EXPECT_NEAR(outVolumeSub, mod * baseG2, 1e-7);
+        EXPECT_NEAR(inVolumeSub, mod * baseG2, 1e-7);
```

Comment:
This does not match the new definition of volume and the test fails accordingly.